File: Assets/Scripts/Core/VotingResult.cs
using System.Collections.Generic;

namespace Bunker.Core
{
    // Describes the outcome of a vote tally. Returned by GameSession so the
    // caller (UI / network layer) can react precisely — show a tie screen,
    // announce an elimination, etc. — instead of inferring it from side effects.
    public class VotingResult
    {
        public enum Outcome
        {
            PlayerEliminated,
            TieRequiresRevote,
            TieUnresolvedNoElimination, // tie persisted after max re-vote attempts
            NoVotesCast
        }

        public Outcome ResultType;
        public PlayerData EliminatedPlayer;          // set only when ResultType == PlayerEliminated
        public List<PlayerData> TiedCandidates;       // set only for tie outcomes
        public Dictionary<string, int> VoteCounts;    // playerId -> vote count, for UI display
    }
}

File: bunker-server/src/game/VotingResult.ts
import { PlayerData } from "./PlayerData";

// Equivalent of the C# VotingResult (introduced fully in section 1.6).
export enum VotingOutcome {
  PlayerEliminated = 0,
  TieRequiresRevote = 1,
  TieUnresolvedNoElimination = 2, // kept for allowVoteTies=true configurations; not reached by default anymore
  NoVotesCast = 3,
  TieResolvedRandomly = 4, // new: forced random elimination among tied candidates after max re-vote attempts
}

export interface VotingResult {
  resultType: VotingOutcome;
  eliminatedPlayer: PlayerData | null;
  tiedCandidates: PlayerData[] | null;
  voteCounts: Map<string, number>;
}
